nautobot/dcim/views.py

```diff
@@ -4362,6 +4393,20 @@ class SoftwareImageFileUIViewSet(NautobotUIViewSet):
         ),
     )
 
+    def get_required_permission(self):
+        # TODO: standardize a pattern for permissions enforcement on custom actions
```

Comment:
Maybe also we could add an `additional_permissions_map` to the UIViewSet base class that could provide this sort of logic?